src/ResourceManagement/Network/NicIPConfigurationImpl.cs

```diff
@@ -166,89 +167,82 @@ internal NicIPConfigurationImpl WithNewPublicIPAddress()
         }
 
 
-        
+
         ///GENMHASH:233A62609A15DC3B8EB48DD8DB699DDC:5F2825180653A2E415C39BF41FD682D6
         internal NicIPConfigurationImpl WithNewPublicIPAddress(string leafDnsLabel)
         {
             return WithNewPublicIPAddress(
                 PrepareCreatablePublicIP(SdkContext.RandomResourceName("pip", 15), leafDnsLabel));
         }
 
-        
+
         ///GENMHASH:BE684C4F4845D0C09A9399569DFB7A42:CDAC048B4CBF3DA15B57B59C55815FF9
         internal NicIPConfigurationImpl WithExistingPublicIPAddress(IPublicIPAddress publicIPAddress)
         {
             return WithExistingPublicIPAddress(publicIPAddress.Id);
         }
 
-        
+
         ///GENMHASH:3C078CA3D79C59C878B566E6BDD55B86:B0EE90118E4A03AFA2010FA1C9E67CB9
         internal NicIPConfigurationImpl WithExistingPublicIPAddress(string resourceId)
         {
             existingPublicIPAddressIdToAssociate = resourceId;
             return this;
         }
 
-        
+
         ///GENMHASH:C4684C8A47F80967DA864E1AB75147B5:D2568543D2B6FA67EEC6DB9538B32535
         internal NicIPConfigurationImpl WithoutPublicIPAddress()
         {
             removePrimaryPublicIPAssociation = true;
             return this;
         }
 
-        
+
         ///GENMHASH:0FBBECB150CBC82F165D8BA614AB135A:055A43C17C98D1AD747257B0659283D6
         internal NicIPConfigurationImpl WithSubnet(string name)
         {
             subnetToAssociate = name;
             return this;
         }
 
-        
-        ///GENMHASH:D36B69B83A3C752672806F0242C22209:FBE8ED51BA892F18B9FBCE3042C8244C
+
+        ///GENMHASH:D36B69B83A3C752672806F0242C22209:760CFCD09BEBB96F67F6C9E6256E6656
         internal NicIPConfigurationImpl WithExistingLoadBalancerBackend(ILoadBalancer loadBalancer, string backendName)
         {
-            foreach (var pool in loadBalancer.Inner.BackendAddressPools)
+            if (loadBalancer != null)
             {
-                if (pool.Name.Equals(backendName, StringComparison.OrdinalIgnoreCase))
+                foreach (var pool in loadBalancer.Inner.BackendAddressPools)
                 {
-                    EnsureBackendAddressPools().Add(pool);
-                    return this;
+                    if (pool.Name.Equals(backendName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        EnsureLoadBalancerBackendAddressPools().Add(pool);
+                        return this;
+                    }
                 }
             }
 
-            return this;
+            return null;
```

Comment:
Returning null here will break Fluent flow, right?

Reply:
well, so that's an intentional change in the logic I made, because the call here is `WithExisting*()` so if it doesn't exist, you shouldn't be able to proceed.  Would throwing an exception (like InvalidOperationException?) be better you think?

Reply:
yes. NullRef exceptions are really not giving any useful info.

Reply:
ok